extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/image_2_text.py

```diff
@@ -0,0 +1,167 @@
+
+        inputs = validate_and_retrieve_image_from_attachments(prompt)
+
+        completion_params["inputs"] = inputs
+
+        await aiconfig.callback_manager.run_callbacks(CallbackEvent("on_deserialize_complete", __name__, {"output": completion_params}))
+        return completion_params
+
+    async def run_inference(self, prompt: Prompt, aiconfig: "AIConfigRuntime", options: InferenceOptions, parameters: Dict[str, Any]) -> list[Output]:
+        await aiconfig.callback_manager.run_callbacks(
+            CallbackEvent(
+                "on_run_start",
+                __name__,
+                {"prompt": prompt, "options": options, "parameters": parameters},
+            )
+        )
+        model_name = aiconfig.get_model_name(prompt)
+
+        self.pipelines[model_name] = pipeline(task="image-to-text", model=model_name)
+
+        captioner = self.pipelines[model_name]
+        completion_data = await self.deserialize(prompt, aiconfig, parameters)
+        print(f"{completion_data=}")
+        inputs = completion_data.pop("inputs")
+        model = completion_data.pop("model")
+        response = captioner(inputs, **completion_data)
+
+        output = ExecuteResult(output_type="execute_result", data=response, metadata={})
+
+        prompt.outputs = [output]
+        await aiconfig.callback_manager.run_callbacks(CallbackEvent("on_run_complete", __name__, {"result": prompt.outputs}))
+        return prompt.outputs
+
+    def get_output_text(self, response: dict[str, Any]) -> str:
+        raise NotImplementedError("get_output_text is not implemented for HuggingFaceImage2TextTransformer")
+
+
+def validate_attachment_type_is_image(attachment: Attachment):
+    if not hasattr(attachment, "mime_type"):
+        raise ValueError(f"Attachment has no mime type. Specify the image mimetype in the aiconfig")
+
+    if not attachment.mime_type.startswith("image/"):
+        raise ValueError(f"Invalid attachment mimetype {attachment.mime_type}. Expected image mimetype.")
+
+
+def validate_and_retrieve_image_from_attachments(prompt: Prompt) -> list[str]:
+    """
+    Retrieves the image uri's from each attachment in the prompt input.
+
+    Throws an exception if
+    - attachment is not image
+    - attachment data is not a uri
+    - no attachments are found
+    - operation fails for any reason
+    """
+
+    if not hasattr(prompt.input, "attachments") or len(prompt.input.attachments) == 0:
+        raise ValueError(f"No attachments found in input for prompt {prompt.name}. Please add an image attachment to the prompt input.")
+
+    image_uris: list[str] = []
+
+    for i, attachment in enumerate(prompt.input.attachments):
+        validate_attachment_type_is_image(attachment)
+
+        if not isinstance(attachment.data, str):
+            # See todo above, but for now only support uri's
+            raise ValueError(f"Attachment #{i} data is not a uri. Please specify a uri for the image attachment in prompt {prompt.name}.")
```

Comment:
Doesn't have to be this diff, but please add support for base64 as well. This is important since if we want to chain prompts, some of our models output in base64 format (ex: text_2_image)

At very least, create an issue to track

Reply:
Fixed in https://github.com/lastmile-ai/aiconfig/pull/856